arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "required": true,
+          "schema": {
+            "$ref": "#/definitions/OperationInputs"
+          }
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IotHubNameAvailabilityInfo"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/listkeys": {
+      "post": {
+        "tags": ["POST"],
+        "summary": "Get all Shared Access Policies for an IotHub",
+        "description": "Get all Shared Access Policies for an IotHub",
+        "operationId": "IotHubResource_ListKeys",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SharedAccessSignatureAuthorizationRuleListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/IotHubKeys/{keyName}/listkeys": {
+      "post": {
+        "tags": ["POST"],
+        "summary": "Get the Key given a specific KeyName for an IotHub",
+        "description": "Get the Key given a specific KeyName for an IotHub",
+        "operationId": "IotHubResource_GetKeysForKeyName",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "name": "keyName",
+          "in": "path",
+          "description": "Name of the key.",
```

Comment:
What are the possible values? "Primary" Or "Secondary"? If so then should this be an enum to make it easy for the end user and should it be named as **keyType** rather than keyName?